torch/_dynamo/variables/functions.py

```diff
@@ -82,6 +82,16 @@ def call_function(
             self, list(self.self_args()) + list(args), kwargs
         )
 
+    def call_hasattr(self, tx, name: str) -> VariableTracker:
+        result = False
+
+        try:
+            result = hasattr(self.get_function(), name)
+        except NotImplementedError:
+            if name == "__name__" and isinstance(self, NestedUserFunctionVariable):
+                result = True
```

Comment:
@voznesenskym do you know if this is legit? What can a NestedUserFunctionVariable be? If it's just a nested `def` or a lambda, then yes: it always has a `__name__`. If it can be something else, then I'm not sure anymore.

Reply:
Answer was yes after some internal discussion

Reply:
@guilhermeleobas can you add a test (maybe to test_misc.py) to test that `.__name__` returns what we expect for a lambda, a def'ed function, and maybe also a test that functools.partial returns something that doesn't have a `__name__`?

Reply:
Done!